src/System.Text.JsonLab.Serialization/System/Text/Json/Serialization/JsonConverter.HandleNumber.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Text.Json.Serialization
+{
+    public static partial class JsonConverter
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static bool HandleNumber(ref Utf8JsonReader reader, JsonConverterSettings options, Type returnType, ref JsonObjectState current)
+        {
+            Debug.Assert(current.PropertyInfo != null);
+
+            Type propType = JsonObjectState.GetElementType(current);
+
+            // More common types are listed first
+
+            if (propType == typeof(int))
+            {
+                if (!reader.TryGetInt32Value(out int value))
+                {
+                    throw new FormatException();
+                }
+                return SetValueAsPrimitive(ref current, value);
```

Comment:
Use the new `GetInt32` APIs that already throw FormatException.

Reply:
Done